Document/0x04f-Testing-Network-Communication.md

```diff
@@ -431,77 +485,73 @@ $ iptables -t nat -A POSTROUTING -j MASQUERADE
 
-- 1st way: Add a [default proxy to the app](https://developer.xamarin.com/api/type/System.Net.WebProxy/ "System.Net.WebProxy Class"), by adding the following code in the `OnCreate` or `Main` method and re-create the app:
+Use @MASTG-TOOL-0076 to achieve a MITM position and redirect port 443 to your interception proxy running on localhost.
 
-    ```cs
-    WebRequest.DefaultWebProxy = new WebProxy("192.168.11.1", 8080);
-    ```
+On macOS:
 
-- 2nd way: Use bettercap in order to get a MITM position, see the section above about how to setup a MITM attack. When being MITM you only need to redirect port 443 to your interception proxy running on localhost. This can be done by using the command `rdr` on macOS:
-
-    ```bash
-    $ echo "
-    rdr pass inet proto tcp from any to any port 443 -> 127.0.0.1 port 8080
-    " | sudo pfctl -ef -
-    ```
+```bash
+echo "
+rdr pass inet proto tcp from any to any port 443 -> 127.0.0.1 port 8080
+" | sudo pfctl -ef -
+```
 
-    For Linux systems you can use `iptables`:
+On Linux:
 
-    ```bash
-    sudo iptables -t nat -A PREROUTING -p tcp --dport 443 -j DNAT --to-destination 127.0.0.1:8080
-    ```
+```bash
+sudo iptables -t nat -A PREROUTING -p tcp --dport 443 -j DNAT --to-destination 127.0.0.1:8080
+```
 
-    As last step, you need to set the option 'Support invisible proxy' in the listener settings of @MASTG-TOOL-0007.
+Lastly, enable **"Support invisible proxy"** in the listener settings of **@MASTG-TOOL-0007**.
 
-- 3rd way: Instead of bettercap an alternative is tweaking the `/etc/hosts` on the mobile phone. Add an entry into `/etc/hosts` for the target domain and point it to the IP address of your intercepting proxy. This creates a similar situation of being MITM as with bettercap and you need to redirect port 443 to the port which is used by your interception proxy. The redirection can be applied as mentioned above. Additionally, you need to redirect traffic from your interception proxy to the original location and port.
+### Option 3: DNS Spoofing
 
-> When redirecting traffic you should create narrow rules to the domains and IPs in scope, to minimize noise and out-of-scope traffic.
+If you can modify the device's DNS resolution ([DNS Spoofing](https://en.wikipedia.org/wiki/DNS_spoofing)), you can reroute the app's traffic to your proxy. For example, on a rooted Android device, you can add an entry in `/etc/hosts` mapping the app's server domain to your proxy machine’s IP. This makes the app believe that your machine is the legitimate server.
 
-The interception proxy need to listen to the port specified in the port forwarding rule above, which is 8080.
+To ensure proper interception, combine DNS spoofing with port redirection. When your machine receives the redirected connection, it will forward the traffic to the proxy. The proxy will then relay the traffic to the real server, effectively acting as a MITM (as done with @MASTG-TOOL-0076).
```

Comment:
You mostly need port redirection for getting the traffic into the proxy, not for the data leaving the proxy. With DNS spoofing, your data will arrive on port 443 (eg) rather than 8080 (eg).